blockchain/miner.py

```diff
@@ -23,8 +23,11 @@ def proof_of_work(last_proof):
     start = timer()
 
     print("Searching for next proof")
-    proof = 0
     #  TODO: Your code here
+    proof = random.randint(0,999999999999)
+    last_hash = hashlib.sha256(f'{last_proof}'.encode()).hexdigest()
```

Comment:
If you are ever in doubt always go back to step 1 of UPER and make sure you understand the problem! You had everything correct about the proof of work and valid proof function except you were hashing the wrong thing.